### ISSUE TITLE
[Carbon 10 🔘 ] Radio Button WCGA 2.1 Guideline 1.4.1 violation  

### ISSUE THREAD

**Issue by Majavita** [carbon-design-system/carbon#8496]
[1.4.1 Use of Color Level A ](https://www.w3.org/WAI/WCAG21/quickref/#identify-purpose) violation by C10 radio button

> Browser

Firefox

## Detailed description

> What version of the Carbon Design System are you using?

v 10.25.0
`carbon-components-react` v 7.25.0

> What did you expect to happen?

"Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element."
We should still see the radio button selected after removal of color or the implementation of another technique:
_"Situation A: If the color of particular words, backgrounds, or other content is used to indicate information:
G14: Ensuring that information conveyed by color differences is also available in text
G205: Including a text cue for colored form control labels
G182: Ensuring that additional visual cues are available when text color differences are used to convey information
G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them"_

> What happened instead?

Button selected versus unselected states look the same after color removal.  

> What WCAG 2.1 checkpoint does the issue violate?

1.4.1 Use of Color Level A 

## Steps to reproduce the issue

Issue can be tested in CDS storybook:
![image](https://user-images.githubusercontent.com/32313977/115890395-83aa1980-a41a-11eb-91fb-a61f23a1d889.png)

In Firefox `Fonts & Colors` settings select System Colors and set "Override the colors specified by the page with your selections above" to "Always"
![image](https://user-images.githubusercontent.com/32313977/115892573-e4d2ec80-a41c-11eb-8f60-1b3bda1b534f.png)

Click one of the radio buttons.  User cannot see button selected.
![image](https://user-images.githubusercontent.com/32313977/115892238-96255280-a41c-11eb-84b9-62a6b5ab840b.png)

>Instructions to remove color may vary between browser.  Additional guidance to configure the browser to ignore colored text:

Firefox®:

Select Tools > Options…
On the Content tab, in the Fonts & Colors section, select the Colors… button.
In the Colors dialog box, select the Use system colors checkbox and deselect the Allow pages to choose their own colors, instead of my selections above checkbox.
Select OK to save changes.
Select OK to exit the Options dialog box.
For Internet Explorer®:

Select Tools >Internet Options.
On the General tab, in the Appearance section, select the Accessibility button.
Check the box Ignore Colors Specified on Web Pages and select OK.
On the General tab, in the Appearance section, select the Colors button and ensure that the checkbox Use Windows colors is selected and select OK to save this setting.
Select Apply and OK to save changes. Note for iOS platform: Mobile Safari does not support removing color from text. Therefore, use a desktop browser to perform the required unit tests.

**Comment by tw15egan** [comment]
cc @dakahn

**Comment by Majavita** [comment]
@joshblack this issue needs to be re-opened per comment in 8497:  There seems to be disagreement on how to satisfy this requirement.  Please refer to https://www.ibm.com/able/requirements/requirements/#1_4_1, and for the instruction for removing color, please refer to https://pages.github.ibm.com/IBMa/able/Test/Manual_tests/web/#configuration-instructions-for-visual-tests.  These components need to be modified in order to pass Carbon Accessibility tests.  Please re-open.

**Comment by dakahn** [comment]
@Majavita You're incorrectly interpreting 1.4.1.

What is the proposed use case for a user going into their browser settings and replacing color with system colors? When and why would this happen? I understand that's how you test 1.4.1, but its not something a user would ever actually do.

In fact it's something _to my knowledge_ you can't even do in Chrome and Edge. That means that 97% of users on the internet can't (and presumably don't need to) use this poorly documented Firefox "accessibility feature".

Okay, so why is that important? Well lets turn our attention back to 1.4.1.

> This requirement ensures that color is not the only visual means of conveying information. Many users have deficiencies in color perception and cannot identify or distinguish colors. Another non-color method for making the information available visually is required, such as with pattern, images, text, or text formatting.

Replacing our colors with system colors does not in any way test whether the perception or ability to distinguish a particular color is required to operate a control. 

So how would I test 1.4.1 compliance for the Carbon radio control? I'd go to the component's documentation and try and see if there were multiple states -- perhaps a warning, or and error state -- and I'd check to make sure those states were communicated not only by color, but also by some iconography or helper text.

A text input with validation that only communicates a user error via a change of the inputs border color to red is an example of a 1.4.1 violation.

An IBM Accessibility requirement 1.4.1 violation
![image](https://user-images.githubusercontent.com/40970507/122259585-f58d6500-ce97-11eb-9442-ed91f1694017.png)

An IBM Accessibility requirement 1.4.1 violation with removed color information
![image](https://user-images.githubusercontent.com/40970507/122259961-574dcf00-ce98-11eb-8a27-309bd0240b75.png)

_Not_ an IBM Accessibility requirement 1.4.1 violation
![image](https://user-images.githubusercontent.com/40970507/122259641-00e09080-ce98-11eb-886a-283e65520318.png)

**Comment by Majavita** [comment]
@dakahn  That same doc also specifies the method for Edge :-).  They just expect the radio button center dot to stay visible.  You disagree with the Accessibility team's directions and therefore with AVT's use of this to open accessibility defects.  
![image](https://user-images.githubusercontent.com/32313977/122293474-a7d72380-cebc-11eb-9e0b-b6897f7de4f0.png)
I have shared your disagreement in the channel already and shall wait for their decision.  Will tag you to the comment.   Other non-Carbon radio buttons and toggles do a better job, but I'll let the Design folks escalate.  Our squad has to use Carbon and we simply need to pass and be compliant.

**Comment by dakahn** [comment]
> That same doc also specifies the method for Edge :-).

Yes, with the use of an external plugin. But without third party tooling those browsers don't even support this use case -- which was the point I was making. 

> Our squad has to use Carbon and we simply need to pass and be compliant.

Completely understand where you're coming from. The reason this issue remains closed is because this testing methodology doesn't actually ensure 1.4.1 compliance. There could be something fundamental I don't understand about 1.4.1 and if that's the case I'm happy to reopen. ☮️
